tests/unit/test_config.py

```diff
@@ -351,21 +351,114 @@ def test_CustomEnum_equalsOp(self):
+		_upgradeConfigFrom_8_to_9_lineIndent(profile)
+		self.assertEqual(profile['documentFormatting']['reportLineIndentation'], 0)
+		with self.assertRaises(KeyError):
+			profile['documentFormatting']['reportLineIndentationWithTones']
+	
+	def test_DefaultProfile_LineIndentationSpeech(self):
+		"""Document formatting Line indentation reporting option set to Speech in default profile."""
+		
+		configString = """
+[documentFormatting]
+	reportLineIndentation = True
+"""
+		profile = _loadProfile(configString)
+		_upgradeConfigFrom_8_to_9_lineIndent(profile)
+		self.assertEqual(profile['documentFormatting']['reportLineIndentation'], 1)
+		with self.assertRaises(KeyError):
+			profile['documentFormatting']['reportLineIndentationWithTones']
+
+	def test_DefaultProfile_LineIndentationTones(self):
+		"""Document formatting Line indentation reporting option set to tones in default profile."""
+		
+		configString = """
+[documentFormatting]
+	reportLineIndentationWithTones = True
+"""
+		profile = _loadProfile(configString)
+		_upgradeConfigFrom_8_to_9_lineIndent(profile)
+		self.assertEqual(profile['documentFormatting']['reportLineIndentation'], 2)
+		with self.assertRaises(KeyError):
+			profile['documentFormatting']['reportLineIndentationWithTones']
+
+	def test_DefaultProfile_LineIndentationSpeechAndTones(self):
+		"""Document formatting Line indentation reporting option set to Speech and tones in default profile."""
+		
+		configString = """
+[documentFormatting]
+	reportLineIndentation = True
+	reportLineIndentationWithTones = True
+"""
+		profile = _loadProfile(configString)
+		_upgradeConfigFrom_8_to_9_lineIndent(profile)
+		self.assertEqual(profile['documentFormatting']['reportLineIndentation'], 3)
+		with self.assertRaises(KeyError):
+			profile['documentFormatting']['reportLineIndentationWithTones']
 
-	@staticmethod
-	def _loadProfile(configString):
-		fn = io.StringIO(configString)
-		profile = configobj.ConfigObj(fn, indent_type="\t", encoding="UTF-8", file_error=False)
-		# Python converts \r\n to \n when reading files in Windows, so ConfigObj can't determine the true line ending.
-		profile.newlines = "\r\n"
-		return profile
+
+class Config_profileUpgradeSteps_upgradeConfigFrom_8_to_9_cellBorders(unittest.TestCase):
 
 	def test_DefaultProfile_Unmodified(self):
-		"""No Braille Show message or Message timeout option modified in default config."""
+		"""Document formatting Cell borders option not modified in default profile."""
+		
+		configString = "[documentFormatting]"
+		profile = _loadProfile(configString)
+		_upgradeConfigFrom_8_to_9_cellBorders(profile)
+		with self.assertRaises(KeyError):
+			profile["documentFormatting"]["reportCellBorders"]
+		with self.assertRaises(KeyError):
+			profile['documentFormatting']['reportBorderStyle']
+		with self.assertRaises(KeyError):
+			profile['documentFormatting']['reportBorderColor']
+
+
+"""[documentFormatting]
+	reportBorderStyle = True
+	reportBorderColor = True
+"""

```

Comment:
This string seems to be unnecessary.

Reply:
> This string seems to be unnecessary.

Sorry, I cannot get from where you commented. The line numbers do not match the ones in the last commit.

Such strings are the config being tested in each function of the unit tests.

Reply:
Hmm, strange. I'm now worried something went wrong when I reviewed this. The changes look quite different now! I'll re-review this tomorrow.